### PR TITLE
Fix coerse scalar list variables

### LINKED ISSUE #13: parseValue doesn't compatible with list of scalars
The main problem that our graphql coerseValue doesn't consider hierarchical structure of input object.

See also https://github.com/tarantool/cartridge/commit/fb4b3a360f8f6d6d34384d0ec8961c7dc4cadf7b

```diff
diff --git a/test/integration/graphql_test.lua b/test/integration/graphql_test.lua
index a4d11e8..0bf9704 100644
--- a/test/integration/graphql_test.lua
+++ b/test/integration/graphql_test.lua
@@ -558,11 +558,25 @@ function g.test_custom_type_scalar_variables()
                 if args.field == nil then
                     return nil
                 end
-                assert(type(args.field) == 'table', "Field is not a table! ")
-                assert(args.field.test ~= nil, "No field 'test' in object!")
+                t.assert_type(args.field, 'table', "Field is not a table! ")
+                t.assert_not_equals(args.field.test, nil, "No field 'test' in object!")
                 return args.field
             end
         },
+        ['test_json_type_list'] = {
+            arguments = {
+                array = types.list(json_type),
+            },
+            kind = types.list(json_type),
+            resolve = function(_, args)
+                if args.array == nil then
+                    return nil
+                end
+                t.assert_type(args.array[1], 'table', "Array element is not a table! ")
+                t.assert_not_equals(args.array[1].test, nil, "No field 'test' in array element!")
+                return args.array
+            end
+        },
         ['test_custom_type_scalar_list'] = {
             kind = types.string,
             arguments = {
@@ -623,6 +637,16 @@ function g.test_custom_type_scalar_variables()
         variables = {},
     }), {test_json_type = 'null'})
 
+    t.assert_equals(check_request([[
+        query($array: [Json]) {
+            test_json_type_list(
+                array: $array
+            )
+        }
+    ]], query_schema, {
+        variables = {array = {json.encode({test = 123})}},
+    }), {test_json_type_list = {'{"test":123}'}})
+
     t.assert_equals(check_request([[
         query($field: CustomString!) {
             test_custom_type_scalar(
```

### PR DESCRIPTION
**Pull request by no1seman**
closes: https://github.com/tarantool/graphql/issues/13

### CODE REVIEW

File: graphql/util.lua [hunk 1]
```diff
@@ -115,6 +115,11 @@ local function coerceValue(node, schemaType, variables, opts)
             node.name.value, variables[node.name.value], schemaType.name
         ))
       end
+    elseif type(value) == 'table' and schemaType.__type == 'List' and
```
Comment (olegrok) [inline graphql/util.lua]: I think something similar should be done for objects.

File: graphql/util.lua [hunk 2]
```diff
@@ -115,6 +115,11 @@ local function coerceValue(node, schemaType, variables, opts)
...
+           type(schemaType.ofType) == 'table' and schemaType.ofType.parseValue ~= nil then
+      for k, v in ipairs(value) do
+      value[k] = schemaType.ofType.parseValue(v)
```
Comment (olegrok) [inline graphql/util.lua]: Indent seems to be missed here

**Review by olegrok — COMMENTED** [review]
It's not deep review, I hope I will see later once again

File: graphql/util.lua (hunk 1 above)
Comment (DifferentialOrange) [reply graphql/util.lua:118]: I've added a simple test: seems that everything is alright

**Review by DifferentialOrange — APPROVED** [review]

File: graphql/util.lua (hunk 1 above)
Comment (DifferentialOrange) [reply graphql/util.lua]: Supported in separate commit

**Pull request merged**